### DISCUSSION TITLE
Need a project init hook.

### DISCUSSION THREAD

**Discussion by maicss** [nuxt/nuxt#24998] [Questions] [answered]
### Describe the feature

Like https://github.com/nuxt/nuxt/discussions/16228 describe, before user interact with our project, we need to check user config, and database connection or database initialization etc. For now, we hardly accomplish it.

1. Should invoked after Nuxt modules and plugins loaded, like database connection, we may need module like `nuxt-mongoose`
2. Should invoked before render or send html to user, user should see `NuxtWelcome` or customized component like `WaitProjectInit`, and wait the hook done.

Thanks.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment, marked as answer]
There are two contexts you should be thinking about.

First, there's build-time context (which is where Nuxt modules exist), and you can use `modules:done` to run at this point after the other modules are loaded. You can use this to load other modules etc.

Second, there's run-time context (which is where plugins run or server routes/HTML is rendered), and for this we have a different set of hooks that you can inject async logic into.

But we very deliberately keep these two contexts separate.

You can see both build-time and run-time hooks here: https://nuxt.com/docs/api/advanced/hooks#app-hooks-runtime
